pkg/scheduler/admission/podnodeconstraints/api/v1/conversion.go

```diff
@@ -0,0 +1,31 @@
+package v1
+
+import (
+	"k8s.io/kubernetes/pkg/conversion"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/util/sets"
+
+	// oapi "github.com/openshift/origin/pkg/api"
+	"github.com/openshift/origin/pkg/scheduler/admission/podnodeconstraints/api"
+)
+
+func convert_v1_PodNodeConstraintsConfig_to_api_PodNodeConstraintsConfig(in *PodNodeConstraintsConfig, out *api.PodNodeConstraintsConfig, s conversion.Scope) error {
+	out.NodeSelectorLabelBlacklist = sets.NewString(in.NodeSelectorLabelBlacklist...)
```

Comment:
This looses stable ordering on the round-trip.  I did this a couple times in the API and it caused pain you'd rather not deal with.  Leave it as a `[]string` and process the value to store the `sets.String` in the admission plugin struct itself: https://github.com/openshift/origin/pull/6979/files#diff-0a00fdcaf1d1642e80ad052899813950R41 .  You'll end up happier.